blockchain/chain_makers.go

```diff
@@ -245,6 +245,20 @@ func GenerateChain(config *params.ChainConfig, parent *types.Block, engine conse
 	return blocks, receipts
 }
 
+// GenerateChainWithGenesis is a wrapper of GenerateChain which will initialize
```

Comment:
Let's not make an arbitrary helper function just for one test. It even has a panic? and the first return value (database.Database) isn't even used.

You can simply do this in the test.
```go
db := database.NewMemoryDBManager()
genesisBlock := genesis.MustCommit(db)
blocks, receipts := GenerateChain(genesis.Config, genesisBlock, engine, db, n, gen)
```